include/config/config.h

```diff
@@ -0,0 +1,258 @@
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ * Copyright (c) 2015 Runtime Inc
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef __SYS_CONFIG_H_
+#define __SYS_CONFIG_H_
+
+#include <misc/util.h>
+#include <misc/slist.h>
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @defgroup sys_config sys/config package
+ * @{
+ */
+
+#define CONF_MAX_DIR_DEPTH	8	/* max depth of config tree */
+#define CONF_MAX_NAME_LEN	(8 * CONF_MAX_DIR_DEPTH)
+#define CONF_MAX_VAL_LEN	256
+#define CONF_NAME_SEPARATOR	"/"
+
+#define CONF_NMGR_OP		0
+
+/**
+ * Type of configuration value.
+ */
+enum conf_type {
+	CONF_NONE = 0,
+	CONF_DIR,
```

Comment:
DIR? does it mean directory? if so it looks odd and does not fit here...also you are not using it your decode functions

Reply:
Agree - doesn't make sense to have this.